src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+// import { error } from 'console';
 
-export class Triangle implements Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const theLongestSide = Math.max(a, b, c);
+    const sumOfThoSmallest = a + b + c - theLongestSide;
+
+    if (a <= 0 || b <= 0 || c <= 0 || sumOfThoSmallest <= theLongestSide) {
+      throw new Error(
+        'Each side ought be >0. And the longest ought be less than others sum',
+      );
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter: number = 0.5 * (this.a + this.b + this.c);
```

Comment:
not an error, but will be more readable
```suggestion
    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
```